CRM/EasyBatch/BAO/EasyBatch.php

```diff
@@ -241,24 +241,40 @@ public static function createAutoFinancialBatch(
       $title .= ts(' non-payment transactions auto') . " {$suffixName}";
       $batchDate = CRM_Utils_Date::processDate(date("Y-m-01"), date("H:i:s"));
     }
-    $params = array(
+
+    // retrieve the batch ID using batch name to avoid create duplicate batch ID
+    $batchID = civicrm_api3('Batch', 'get', [
+      'name' => substr(str_replace(' ', '_', $title), 0, 64),
```

Comment:
batch name is trimmed to 64 as per batch.name length attribute [here](https://github.com/civicrm/civicrm-core/blob/master/xml/schema/Batch/Batch.xml#L28)